CHANGELOG.md

```diff
@@ -1,3 +1,319 @@
+
+##### Bare-metal
+
+The `cluster.os_channel` parameter got simplified by removing the `flatcar-` prefix.
+
+###### Old
+
+```hcl
+os_channel = "flatcar-stable"
+```
+
+###### New
+
+```hcl
+os_channel = "stable"
+```
+
+##### Velero
+
+Velero requires an explicit `provider` field to select the provider.
+Example:
+
+```hcl
+component `velero` {
+  provider = "openebs"
+
+  openebs {
+    ...
+  }
+}
+```
+
+#### Updating Prometheus Operator
+
+Steps (1-4) must be performed before updating the cluster i.e. before running `lokoctl cluster apply`.
+
+Steps (5-6) must be performed after updating the cluster i.e. after running `lokoctl cluster apply`.
+
+Upgrading this component incurs down time and some steps to be executed manually before upgrading:
+
+1. Patch the `PersistentVolume` created/used by the `prometheus-operator` component to `Retain` claim policy.
+
+  ```bash
+  kubectl patch pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-prometheus-prometheus-operator-prometheus-0")].metadata.name}')
+
+  kubectl patch pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-alertmanager-prometheus-operator-alertmanager-0")].metadata.name}')
+  ```
+
+  > **Note:** To execute the above command, the user must have a cluster wide permission.
+
+2. Uninstall the `prometheus-operator` release and delete the existing `PersistentVolumeClaim`, and verify `PersistentVolume` become `Released`.
+
+  ```bash
+  lokoctl component delete prometheus-operator
+  kubectl delete pvc data-prometheus-prometheus-operator-prometheus-0 -n monitoring
+  kubectl delete pvc data-alertmanager-prometheus-operator-alertmanager-0 -n monitoring
+  ```
+
+  You can choose to remove all your existing CRDs (ServiceMonitors, PodMonitors, etc.) if you want to.
+
+3. Remove current `spec.claimRef` values to change the PV's status from Released to Available.
+
+  ```bash
+  kubectl patch pv --type json -p='[{"op": "remove", "path": "/spec/claimRef"}]' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-prometheus-prometheus-operator-prometheus-0")].metadata.name}')
+
+  kubectl patch pv --type json -p='[{"op": "remove", "path": "/spec/claimRef"}]' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-alertmanager-prometheus-operator-alertmanager-0")].metadata.name}')
+  ```
+
+  > **Note:** To execute the above command, the user must have a cluster wide permission.
+
+4. Make sure that the `prometheus-operator`'s `storage_class` and `prometheus.storage_size` are unchanged during the upgrade process.
+
+5. Proceed to a fresh `prometheus-operator` component installation. The new release should now re-attach your previously released PV with its content.
+
+```
+lokoctl component apply prometheus-operator
+```
```

Comment:
@ipochi we suggest that the prometheus operator to be installed only after cluster is updated. But the cluster update process is really long for this release. Can't we suggest to user that they install newer prometheus-operator before update?

Reply:
Ideally no, because we emphasize that Lokomotive control plane must be updated first and what follows is Lokomotive components.

That being said for instructing to update the component first, it would need to be tested if everything comes up fine, post control plane upgrade.

Reply:
What other problem do you foresee apart from the cluster update process being lengthy ?

Reply:
The script that updates the nodes is modifying kubelet, etcd configs and restarting them. This change happens one node at a time hence it is lengthy (the running time can vary depending on the cluster size). Then comes the control plane update part.

Anything going wrong during this period, user has no way of know what hit them or how to mitigate the issue faster because they have no idea at what level things stopped working.

Reply:
Ok and that would be classified as maintenance window right ? What difference it would make to the process if the Prometheus operator component is updated before or after.

Wouldn't the user have to deal with addtional overhead of an updated prometheus operator if something goes wrong in the update process ?

Reply:
> Ok and that would be classified as maintenance window right?

Yes I guess that would be counted as maintenance window but there are too many moving parts here.

> What difference it would make to the process if the Prometheus operator component is updated before or after.

Updating component after the cluster update: the release notes from the PR for prometheus operator update process said that certain steps should be performed before cluster update. The new component is installed only after cluster is updated. I think we do that PV patching to make sure that data is not lost due to prometheus-operator component update (especially due to the change of names).

Updating component before cluster update: yes that the question I am interested in, why can't we do all of it before?

> Wouldn't the user have to deal with addtional overhead of an updated prometheus operator if something goes wrong in the update process ?

Here we come into picture. We figure out all the stuff that can go wrong. Provide mitigation plans and if nothing works user can still rollback to older `prometheus-operator` component, unlike the updating cluster where rollback would be much much harder.

Reply:
I don't have any issues with updating the prometheus operator component beore the cluster update, provided this process is tested and there are no consequences.

Reply:
>The script that updates the nodes is modifying kubelet, etcd configs and restarting them. This change happens one node at a time hence it is lengthy (the running time can vary depending on the cluster size). Then comes the control plane update part.

Controlplane should be upgraded first, then nodes. This is how Kubernetes version skew recommends it.

About the Prometheus Operator, IMO it should be updated after the controlplane, as we don't test new Prometheus version with old Kubernetes. The same applies for all other components.

Reply:
I am addressing the updating kubelet etcd and kubelet version after control plane has updated. But I still think that prometheus-operator should be updated before the cluster update.